hypershift-operator/controllers/hostedcluster/apiproviders/platform_reconciler_aws.go

```diff
@@ -0,0 +1,376 @@
+								DefaultMode: &defaultMode,
+								SecretName:  "capi-webhooks-tls",
+							},
+						},
+					},
+					{
+						Name: "credentials",
+						VolumeSource: corev1.VolumeSource{
+							Secret: &corev1.SecretVolumeSource{
+								SecretName: manifests.AWSNodePoolManagementCreds(deployment.Namespace).Name,
+							},
+						},
+					},
+				},
+				Containers: []corev1.Container{
+					{
+						Name:            "manager",
+						Image:           imageCAPA,
+						ImagePullPolicy: corev1.PullAlways,
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      "credentials",
+								MountPath: "/home/.aws",
+							},
+							{
+								Name:      "capi-webhooks-tls",
+								ReadOnly:  true,
+								MountPath: "/tmp/k8s-webhook-server/serving-certs",
+							},
+						},
+						Env: []corev1.EnvVar{
+							{
+								Name: "MY_NAMESPACE",
+								ValueFrom: &corev1.EnvVarSource{
+									FieldRef: &corev1.ObjectFieldSelector{
+										FieldPath: "metadata.namespace",
+									},
+								},
+							},
+							{
+								Name:  "AWS_SHARED_CREDENTIALS_FILE",
+								Value: "/home/.aws/credentials",
+							},
+						},
+						Command: []string{"/manager"},
+						Args: []string{"--namespace", "$(MY_NAMESPACE)",
+							"--alsologtostderr",
+							"--v=4",
+							"--leader-elect=true",
+							"--feature-gates=EKS=false",
+						},
+						Ports: []corev1.ContainerPort{
+							{
+								Name:          "healthz",
+								ContainerPort: 9440,
+								Protocol:      corev1.ProtocolTCP,
+							},
+						},
+						LivenessProbe: &corev1.Probe{
+							Handler: corev1.Handler{
+								HTTPGet: &corev1.HTTPGetAction{
+									Path: "/healthz",
+									Port: intstr.FromString("healthz"),
+								},
+							},
+						},
+						ReadinessProbe: &corev1.Probe{
+							Handler: corev1.Handler{
+								HTTPGet: &corev1.HTTPGetAction{
+									Path: "/readyz",
+									Port: intstr.FromString("healthz"),
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	hyperutil.SetColocation(hc, deployment)
+	// TODO (alberto): Reconsider enable this back when we face a real need
+	// with no better solution.
+	// hyperutil.SetRestartAnnotation(hc, deployment)
+	hyperutil.SetControlPlaneIsolation(hc, deployment)
+	hyperutil.SetDefaultPriorityClass(deployment)
+	switch hc.Spec.ControllerAvailabilityPolicy {
+	case hyperv1.HighlyAvailable:
+		maxSurge := intstr.FromInt(1)
+		maxUnavailable := intstr.FromInt(1)
+		deployment.Spec.Strategy.Type = appsv1.RollingUpdateDeploymentStrategyType
+		deployment.Spec.Strategy.RollingUpdate = &appsv1.RollingUpdateDeployment{
+			MaxSurge:       &maxSurge,
+			MaxUnavailable: &maxUnavailable,
+		}
+		deployment.Spec.Replicas = k8sutilspointer.Int32Ptr(3)
+		hyperutil.SetMultizoneSpread(capaLabels, deployment)
+	default:
+		deployment.Spec.Replicas = k8sutilspointer.Int32Ptr(1)
+	}
+
+	return nil
+}
+
+func reconcileCAPIAWSProviderRole(role *rbacv1.Role) error {
```

Comment:
This function content not look like AWS specific code, this can be used also for other platforms
@enxebre what do you say?
Also the same comment to reconcileCAPIAWSProviderRoleBinding, and might be others functions